src/App.jsx

```diff
@@ -14,33 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const REVERSE_SORT = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [reversed, setReversed] = useState('');
+  const [lightlyClass, setLightlyClass] = useState('');
+  const [lightlyClassReversed, setlightlyClassReversed] = useState(false);
+  let [...visibleGoods] = goodsFromServer;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  visibleGoods = visibleGoods.sort((good1, good2) => {
+    switch (sortBy) {
+      case SORT_BY_ALPHABET:
+        return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed === REVERSE_SORT) {
+    visibleGoods.reverse();
+  }
+
+  function toggleReverse() {
+    if (reversed === REVERSE_SORT) {
+      setReversed('');
+      setlightlyClassReversed(false);
+    } else {
+      setReversed(REVERSE_SORT);
+      setlightlyClassReversed(true);
+    }
```

Comment:
If reversed is a boolean value, then we can set the opposite value and avoid using `if else` block here